x-pack/solutions/search/plugins/search_homepage/public/hooks/use_ingestion_cta_feature_flag.ts

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { SEARCH_HOMEPAGE_INGESTION_CTA_FEATURE_FLAG } from '@kbn/search-shared-ui/src/constants';
+import { useKibana } from './use_kibana';
+
+export const useIngestionCtaFeatureFlag = (): boolean => {
+  const { cloud, featureFlags } = useKibana().services;
+  if (cloud?.isServerlessEnabled !== true) return false;
+
+  return featureFlags.getBooleanValue(SEARCH_HOMEPAGE_INGESTION_CTA_FEATURE_FLAG, false);
```

Comment:
nitpick:
```suggestion
  return cloud?.isServerlessEnabled && featureFlags.getBooleanValue(SEARCH_HOMEPAGE_INGESTION_CTA_FEATURE_FLAG, false);
```

Reply:
I think it's more readable with the `if` statement in this situation. as `isServerlessEnabled` check would need `!!` before it to handle undefined case.